### PR TITLE
ConsoleInteraction.py: Restore format_str

### PR DESCRIPTION
**Pull request by Mixih**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by Mixih** [comment]
nto sure whyt his pull isn't linking, but here is the xref: https://github.com/coala/coala/issues/3303

**Comment by jayvdb** [comment]
In commit message, "Readd" should be "Re-add" for readability, or "Restore" is another option.

4db26a3 has seemingly unrelated PEP-ish changes like
```diff
-                no_color, line[:sourcerange.start.column-1], lexer), end='')
+                no_color, line[:sourcerange.start.column - 1], lexer), end='')
```

Why are they needed as part of this?  as this is to be backported, we need to be careful about unnecessary changes.

**Comment by Mixih** [comment]
that's a separate commit....

**Comment by Mixih** [comment]
to clarify: back only requested for ea7c5ce . The other one is a PEP8 maintenance patch...

File: coalib/output/ConsoleInteraction.py [hunk 1]
```diff
@@ -278,15 +278,26 @@ def print_diffs_info(diffs, printer):
             color='green')
 
 
+_warn_deprecated_format_str = True  # Remove when format_str is deprecated
```
Comment (jayvdb) [inline coalib/output/ConsoleInteraction.py]: The stdlib warnings module does this for you.

**Comment by jayvdb** [comment]
ack 051d3b2
ack 3ff6cf3

File: coalib/output/ConsoleInteraction.py (hunk 1 above)
Comment (Mixih) [inline coalib/output/ConsoleInteraction.py]: Hm, but that wouldn't use our LogPrinter facility provided though the parameters... Are we willing to make that tradeoff?

Comment (jayvdb) [inline coalib/output/ConsoleInteraction.py]: nope; I ack'd, as this isnt the bug to start using `warnings`.  It works; lets get it done.

**Comment by Mixih** [comment]
@jayvdb your ack didn't register

**Comment by jayvdb** [comment]
you need to rebase anyway.

**Comment by Mixih** [comment]
Rebase done

**Comment by jayvdb** [comment]
ack 17cd716
ack d05a091

**Comment by Mixih** [comment]
reack 
17cd716
d05a091

**Comment by Mixih** [comment]
ack  	17cd716

**Comment by jayvdb** [comment]
reack 17cd716

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**